central/declarativeconfig/health/datastore.go

```diff
@@ -0,0 +1,25 @@
+package health
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/declarativeconfig/health/store"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// DataStore is the entry point for modifying declarative config health data.
+//
+//go:generate mockgen-wrapper
+type DataStore interface {
+	GetDeclarativeConfigs(ctx context.Context) ([]*storage.DeclarativeConfigHealth, error)
+	UpsertDeclarativeConfig(ctx context.Context, configHealth *storage.DeclarativeConfigHealth) error
+	RemoveDeclarativeConfig(ctx context.Context, id string) error
+	GetDeclarativeConfig(ctx context.Context, id string) (*storage.DeclarativeConfigHealth, bool, error)
```

Comment:
Let's see when we refactor the existing code to use the new datastore, but this could be potentially unused since in the current implementation it's also not used.

Reply:
True - let's remove it after we implement logic on top of this PR